sw/device/lib/uart.c

```diff
@@ -42,5 +42,13 @@ void uart_send_uint(uint32_t n, int bits) {
 }
 
 int uart_rcv_char(char *c) {
-  return dif_uart_byte_receive_polled(&uart0, (uint8_t *)c);
+  size_t num_bytes = 0;
+  if (dif_uart_rx_bytes_available(&uart0, &num_bytes) == false) {
+    return -1;
+  }
+  if (num_bytes == 0) {
+    return -1;
+  }
+  return (dif_uart_bytes_receive(&uart0, 1, (uint8_t *)c, NULL) == true) ? 0
```

Comment:
``` return dif_uart_bytes_receive(&uart0, 1, (uint8_t *)c, NULL) ? 0 : -1```

The return from `dif_uart_bytes_receive` is already `bool`.

Reply:
This function returns `int`, not `bool`, so the ternary converts from bool back to the 2-valued int. Yes the int should really have been a bool, but let's not update the API now.

Reply:
I don't want to change the return type here, I just want to use the implicit booleanity of `dif_uart_bytes_receive`, just removing the `== true`.

Reply:
Oh, my mistake. Yes this needs correcting.

Reply:
Thanks @gkelly for your feedback. I changed this as suggested.